extensions/wikia/UserLogin/js/FacebookLogin.js

```diff
@@ -287,6 +287,30 @@
 					}
 				}
 			});
+
+			this.initSignupFormValidation();
+		},
+
+		initSignupFormValidation: function () {
+			var validator,
+				wikiaForm = this.signupForm.wikiaForm,
+				inputs = wikiaForm.inputs,
+				inputsToValidate = ['username', 'password'];
+
+			if (inputs.email) {
+				inputsToValidate.push('email');
+			}
+
+			validator = new UserSignupAjaxValidation({
+				wikiaForm: wikiaForm,
+				inputsToValidate: inputsToValidate,
+				submitButton: inputs.submit
+			});
+
+			// Add validation on blur event for all inputs to validate
+			inputsToValidate.forEach(function (inputName) {
+				inputs[inputName].on('blur', validator.validateMappedInput.bind(validator));
```

Comment:
I donno, maybe I just can't think of a good way to do this, but it's awkward b/c the second parameter in `$.fn.on` is a string selector, not a jQuery object. So I could theoretically do something like:

```javascript
selector = '[name=' + inputsToValidate.join('], [name=') + ']';
```
Then I could pass that `selector` string as the second argument like:

```javascript
wikiaForm.el.on('blur', selector, validate));
```
That's the only way I can see to make it more performant, b/c I would be attaching the event listener to the parent element as opposed to the inputs themselves. 

To me though, that code look brittle and unclear to the next person that come across it. In theory it would be nice to attach only one event listener but in practice I think it's fine to leave this as is. @rafalkalinski, could you think of a better way to handle this?